src/port1.0/portconfigure.tcl

```diff
@@ -1367,27 +1367,23 @@ proc portconfigure::configure_main {args} {
     foreach flags {cflags cppflags cxxflags objcflags objcxxflags ldflags fflags f90flags fcflags} {
         global configure.${flags} configure.universal_${flags}
     }
-
-    if {[tbool use_autoreconf]} {
-        if {[catch {command_exec autoreconf} result]} {
-            return -code error "[format [msgcat::mc "%s failure: %s"] autoreconf $result]"
-        }
-    }
-
-    if {[tbool use_automake]} {
-        if {[catch {command_exec automake} result]} {
-            return -code error "[format [msgcat::mc "%s failure: %s"] automake $result]"
-        }
-    }
-
-    if {[tbool use_autoconf]} {
-        if {[catch {command_exec autoconf} result]} {
-            return -code error "[format [msgcat::mc "%s failure: %s"] autoconf $result]"
+    # Export flags into destroot and build
+    foreach phase {destroot build} {
+        parse_environment ${phase}
+        append_to_environment_value ${phase} DEVELOPER_DIR {*}[option configure.developer_dir]
+    }
+    # If enabled, run autoreconf/automake/autoconf
+    foreach config {autoreconf automake autoconf} {
+        if {[tbool use_${config}]} {
+            append_to_environment_value ${config} DEVELOPER_DIR {*}[option configure.developer_dir]
```

Comment:
If the goal is to always add DEVELOPER_DIR to the environment, there is [precedent](https://github.com/macports/macports-base/blob/6f8aca0267707bd4d8e122148e3f88af57029a73/src/port1.0/portutil.tcl#L417) for that sort of thing in `command_exec`. Perhaps we can do something similar.

Reply:
Right, that looks much cleaner. I've created the solution which adds the DEVELOPER_DIR in `command_exec` here https://github.com/satraul/macports-base/commit/ad8df129ed674fee3e532fe8b7eb69f5afbc4b6f. It seems both solutions aren't working yet tho.. I'm testing both on emptyport with `build { system -W "/" "clang -v"}` but it's still going to Xcode.

Reply:
If you're running something with `system` (or `exec`) directly, that's completely separate from all the `command` machinery. It's the caller's responsibility to set any required environment variables in that case.